test/test_eos_unit.cpp

```diff
@@ -64,6 +66,38 @@ constexpr Real ev2k = 1.160451812e4;
 #endif // SINGULARITY_TEST_SESAME
 #endif // SPINER_USE_HDF
 
+template <typename E1, typename E2>
+inline void compare_two_eoss(E1 &&test_e, E2 &&ref_e) {
+  // compare all individual member functions with 1 as inputs,
+  // this function is meant to catch mis-implementations of
+  // modifiers that can be initialized in such a way as to
+  // be equivalent of an unmodified eos. Best used with analytic
+  // eoss.
+  REQUIRE(isClose(test_e.TemperatureFromDensityInternalEnergy(1, 1),
+                  ref_e.TemperatureFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.InternalEnergyFromDensityTemperature(1, 1),
+                  ref_e.InternalEnergyFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.PressureFromDensityInternalEnergy(1, 1),
+                  ref_e.PressureFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.SpecificHeatFromDensityInternalEnergy(1, 1),
+                  ref_e.SpecificHeatFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.BulkModulusFromDensityInternalEnergy(1, 1),
+                  ref_e.BulkModulusFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.GruneisenParamFromDensityInternalEnergy(1, 1),
+                  ref_e.GruneisenParamFromDensityInternalEnergy(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.PressureFromDensityTemperature(1, 1),
+                  ref_e.PressureFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.SpecificHeatFromDensityTemperature(1, 1),
+                  ref_e.SpecificHeatFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.BulkModulusFromDensityTemperature(1, 1),
+                  ref_e.BulkModulusFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.GruneisenParamFromDensityTemperature(1, 1),
+                  ref_e.GruneisenParamFromDensityTemperature(1, 1), 1.e-15));
+  REQUIRE(isClose(test_e.MinimumDensity(), ref_e.MinimumDensity(), 1.e-15));
+  REQUIRE(isClose(test_e.MinimumTemperature(), ref_e.MinimumTemperature(), 1.e-15));
```

Comment:
This isn't a hold on the MR, but one thing I've been trying to do is use the Catch2 `INFO` function since `REQUIRE(isClose(...))` will just return `false` without any more helpful information. The `INFO` macro is a print statement that will execute only when the test fails so you can get some more information about why it's failing. This has _massively_ helped in my debugging of the Gruneisen EOS because just knowing the two values can help me determine what's going crazy.

The other thing to consider is using `CHECK` rather than `REQUIRE` for a big block like this. `CHECK` fails the test still if the condition is false, but it won't end the test at the first failure. This can be helpful for debugging when you can see a pattern in how multiple values are off.

Reply:
done.